Mathlib/LinearAlgebra/FiniteDimensional.lean

```diff
@@ -1179,11 +1179,13 @@ theorem Subalgebra.eq_bot_of_rank_le_one {S : Subalgebra F E} (h : Module.rank F
   exact h.trans_eq Subalgebra.rank_bot.symm
 #align subalgebra.eq_bot_of_rank_le_one Subalgebra.eq_bot_of_rank_le_one
 
-theorem Subalgebra.eq_bot_of_finrank_one {S : Subalgebra F E} (h : finrank F S = 1) : S = ⊥ :=
-  Subalgebra.eq_bot_of_rank_le_one <| by
```

Comment:
Can you generalize the proof of `Subalgebra.eq_bot_of_rank_le_one` instead?

Reply:
Let me see.

Reply:
I suspect that starting from `finrank_eq_one_iff` the conditon can be relaxed. Let me try.

Reply:
Done.